docs/src/content/docs/book/functions.mdx

```diff
@@ -1,184 +1,463 @@
-  let res: Int = 1;
-  repeat(c) {
-    res *= a;
-  }
-  return res;
+fun add(a: Int, b: Int): Int {
+    return a + b;
 }
 ```
 
-## Virtual and abstract functions
+Read about common aspects of functions: [Commonalities](#commonalities).
+
+### Global functions {#fun-global}
 
-You can allow a contract inheriting a [trait](/book/types#traits) to modify an internal function marked with the `virtual{:tact}` keyword by using `override{:tact}`. A function can also be marked as `abstract{:tact}`, in which case the inheriting contract must define its implementation:
+<Badge text="Global scope" variant="note" size="medium"/><p/>
+
+Regular functions that can be defined only at the top (module) level are called global. They differ from [internal functions](#fun-internal) not only in the scope, but also in available [attributes](#attributes).
 
 ```tact
-trait FilterTrait with Ownable {
-    // Virtual functions can be overridden by users of this trait
-    virtual fun filterMessage(): Bool {
-        return sender() != self.owner;
+fun customPow(a: Int, c: Int): Int {
+    let res: Int = 1;
+    repeat (c) {
+        res *= a;
     }
-
-    abstract fun specialFilter(): Bool;
+    return res;
 }
+```
 
-contract Filter with FilterTrait {
-    // Overriding the default behavior of the FilterTrait
-    override fun filterMessage(): Bool {
-        return true;
-    }
+The following attributes can be specified for global functions:
 
-    override fun specialFilter(): Bool {
-        return true;
+* [`inline{:tact}`](#attributes-inline) — embeds the function contents at the call site.
+* [`extends{:tact}`](#attributes-extends) — makes it an [extension function](#extends).
+* [`mutates{:tact}`](#attributes-mutates), along with [`extends{:tact}`](#attributes-extends) — makes it an [extension mutation function](#mutates).
+
+These attributes _cannot_ be specified:
+
+* [`abstract{:tact}`](#attributes-abstract), [`virtual{:tact}`](#attributes-virtual) and [`override{:tact}`](#attributes-override) — global functions cannot be defined within a contract or a trait.
+* [`get{:tact}`](#attributes-get) — global functions cannot be [getters](#get).
+
+### Internal functions {#fun-internal}
+
+<Badge text="Contract scope" variant="note" size="medium"/><p/>
+
+These functions behave similarly to private methods in popular object-oriented languages — they are internal to contracts and can be called by prefixing their names with a special [identifier `self{:tact}`](/book/contracts#self). That is why internal (or member) functions are sometimes referred to as "contract methods".
+
+Internal functions can access the contract's [persistent state variables](/book/contracts#variables) and [constants](/book/contracts#constants).
+
+They can only be called from [receivers](#receivers), [getters](#get), and other internal functions, but not from other contracts.
+
+```tact
+contract InternalFun() {
+    fun add(a: Int, b: Int): Int {
+        return a + b;
```

Comment:
let's make this function use a contract parameter, so `self` actually makes sense below